src/highdicom/sr/templates.py

```diff
@@ -2918,13 +2925,65 @@ def roi(self) -> Union[ImageRegion, ImageRegion3D, None]:
         if len(matches) > 1:
             logger.warning(
                 'found more than one "Image Region" content item '
-                'in "Planar ROI Measurements and Qualitative Evalutions" '
+                'in "Planar ROI Measurements and Qualitative Evaluations" '
                 'template'
             )
-        if len(matches) == 0:
+        if len(matches) > 0:
             return ImageRegion3D.from_dataset(matches[0])
         return None
 
+    @property
+    def referenced_segment(
+        self
+    ) -> Union[ImageContentItem, None]:
+        """Union[highdicom.sr.ImageContentItem, None]:
+        segmentation frame referenced by the measurements group
+        """  # noqa: E501
+        root_item = self[0]
+
+        # Find the referenced segmentation frame content item
+        matches = find_content_items(
+            root_item,
+            name=codes.DCM.ReferencedSegmentationFrame,
+            value_type=ValueTypeValues.IMAGE
+        )
+        if len(matches) > 1:
+            logger.warning(
+                'found more than one "Referenced Segmentation Frame" content '
+                'item in "Planar ROI Measurements and Qualitative Evaluations" '
+                'template'
+            )
+        elif len(matches) == 0:
+            return None
+
+        return ImageContentItem.from_dataset(matches[0])
+
+    @property
+    def source_image_for_segmentation(
+        self
+    ) -> Union[SourceImageForSegmentation, None]:
+        """Union[highdicom.sr.SourceImageForSegmentation, None]:
+        source images used for the referenced segment
+        """  # noqa: E501
+        root_item = self[0]
+
+        # Find the referenced segmentation frame content item
+        matches = find_content_items(
+            root_item,
+            name=codes.DCM.SourceImageForSegmentation,
+            value_type=ValueTypeValues.IMAGE
+        )
+        if len(matches) > 1:
+            logger.warning(
+                'found more than one "Source Image For Segmentation" content '
+                'item in "Planar ROI Measurements and Qualitative Evaluations" '
+                'template'
+            )
+        elif len(matches) == 0:
+            return None
+
+        return SourceImageForSegmentation.from_dataset(matches[0])
+
```

Comment:
I think it would be cleaner to return `Union[hd.seg.ReferencedSegment, hd.seg.ReferencedSegmentationFrame]` so that the segmentation and corresponding source images are referenced together and match the constructor arguments.

Reply:
After refactoring, this now returns a `Union[ReferencedSegmentationFrame, None]`. Note that a `PlanarROIMeasurementsAndQualitativeEvaluations` cannot contain a "Referenced Segment" (they can only be used in the volumetric measurements group)